doc/manual/documentation.rst

```diff
@@ -119,9 +121,26 @@ documentation:
        ```
        """
 
-   You can then run ``make -C doc doctest`` to run all the doctests, which will ensure
-   that your example works. Note that whitespace in your doctest is significant! The
-   doctest will fail if you misalign the output of pretty-printing an array, for example.
+   .. warning::
+
+      Calling ``rand`` and other RNG-related functions should be avoided in doctests since
+      they will not produce consistent outputs during different Julia sessions. Printing the
+      contents of ``Dict``\ s, ``Set``\ s, or any other unordered types should also be
```

Comment:
depends on what generates the inputs, but I see this vary usually because of word size differences in hashing, and we already have architecture specificity in Int size and paths and a few other aspects of doctest output